src/ipc/ipc4/handler.c

```diff
@@ -537,7 +537,8 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 		if (ret)
 			tr_err(&ipc_tr, "failed to create chain dma %d", ret);
 
-		return ret;
+		if (!cdma.primary.r.enable || ret)
+			return ret;
```

Comment:
For future PRs, please add a bit more text in git commit to explain the problem and why the fix helps. In what case the old code did not work, and how the patch addresses the problem. This makes the review process easier.

Reply:
ack - can we have an inline comment here. Difficult to follow.